### PR TITLE
Add basic Xamarin.iOS support

### PR DESCRIPTION
**Pull request by rmarinho**
Basic work to show usage on Xamarin iOS , contains implementation of YogaKit and sample project.

Should help on #276

### CODE REVIEW

**Comment by emilsjolander** [comment]
@splhack Can you have a look?

File: csharp/Facebook.Yoga/ios/CustomBuildAction.targets [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+	<PropertyGroup>
+	    <CompileDependsOn>CopyInNativeLib;$(CompileDependsOn)</CompileDependsOn>
+	</PropertyGroup>
+
+	<!-- The # and , in this path does not play nice with the binding project logic, so make a copy -->
+	<Target Name="CopyInNativeLib" Inputs="..\..\..\buck-out\gen\yoga#default,static\libyoga.a" Outputs="$(ProjectDir)libyoga.a">
```
Comment (splhack) [inline csharp/Facebook.Yoga/ios/CustomBuildAction.targets]: With #286, we can use the fat static library `buck-out/gen/csharp/yoganet-ios/libyoga.a` that includes iOS simulator 64bit, arm64 and armv7.

File: .gitignore [hunk 2]
```diff
@@ -8,3 +8,4 @@
 
 # Visual studio code
 .vscode
+libyoga.a
```
Comment (splhack) [inline .gitignore]: Move to csharp/.gitignore

File: csharp/Facebook.Yoga/ios/CustomBuildAction.targets (hunk 1 above)
Comment (rmarinho) [inline csharp/Facebook.Yoga/ios/CustomBuildAction.targets]: it didn't worked for me the 1st time.. i m going try again..

**Comment by rmarinho** [comment]
Yap, i agree with the new hierarchy, will change.

File: csharp/Facebook.Yoga/ios/CustomBuildAction.targets (hunk 1 above)
Comment (rmarinho) [inline csharp/Facebook.Yoga/ios/CustomBuildAction.targets]: Oh right, i pulled the new PR that does the fat lib and it works, we will need to get #286 in.

**Comment by rmarinho** [comment]
All changes were made i think. Ready for another review.

**Comment by rmarinho** [comment]
@splhack Can you take another look, next step after get this in is to work on a Nuget

**Comment by emilsjolander** [comment]
@rmarinho It is holiday season so this might take a while to get fully reviews and merged 👍

**Comment by splhack** [comment]
@rmarinho I'm thinking to merge #283 first. Do you mind to take a look at #283? I think Xamarin.iOS can use `csharp/Facebook.Yoga/Facebook.Yoga.Shared.shproj` as well (Add `Facebook.Yoga.Shared` into `Facebook.Yoga.iOS` References)

**Comment by rmarinho** [comment]
Yap seems good, i will rebase and make changes once that one is in.

**Comment by emilsjolander** [comment]
Want to make sure you are aware of changes being made to the iOS wrapper currently as it might effect this PR https://github.com/facebook/yoga/pull/289

**Comment by emilsjolander** [comment]
https://github.com/facebook/yoga/pull/283 has been merged now

**Comment by rmarinho** [comment]
yap going to work on this tonight.

**Pull request closed without merging**